src/subscriptions.rs

```diff
@@ -1,20 +1,248 @@
-use bitcoin_hashes::sha256d::Hash as Sha256dHash;
+use bitcoin_hashes::{sha256d::Hash as Sha256dHash, Hash};
 use bitcoin_hashes::hex::FromHex;
+use bitcoin::blockdata::transaction::Transaction;
+use bitcoin::hash_types::{BlockHash, Txid};
 
 use rusoto_core::Region;
 use rusoto_dynamodb::{DynamoDb, DynamoDbClient, ScanInput};
+use rusoto_sqs::{ReceiveMessageRequest, Sqs, SqsClient, DeleteMessageRequest, GetQueueUrlRequest, GetQueueUrlResult, SendMessageRequest};
 
 use std::default::Default;
 use std::env;
 use std::format;
+use std::sync::{Arc};
+use std::sync::mpsc::{SyncSender};
+use std::collections::{HashMap, HashSet};
+use serde_json::Value;
+use std::time::Instant;
 
 use crate::errors::*;
-use std::collections::HashMap;
-use serde_json::Value;
+use crate::index::compute_script_hash;
+use crate::query::{Query};
+use crate::util::{spawn_thread, SyncChannel, HeaderEntry, FullHash};
+
+fn get_output_scripthash(txn: &Transaction, n: Option<usize>) -> Vec<FullHash> {
+    if let Some(out) = n {
+        vec![compute_script_hash(&txn.output[out].script_pubkey[..])]
+    } else {
+        txn.output
+            .iter()
+            .map(|o| compute_script_hash(&o.script_pubkey[..]))
+            .collect()
+    }
+}
+
+fn hash_from_value<T: Hash>(val: Option<&Value>) -> Result<T> {
+    let script_hash = val.chain_err(|| "missing hash")?;
+    let script_hash = script_hash.as_str().chain_err(|| "non-string hash")?;
+    let script_hash = T::from_hex(script_hash).chain_err(|| "non-hex hash")?;
+    Ok(script_hash)
+}
+
+#[derive(Debug)]
+pub enum SubscriptionMessage {
+    NewScriptHash(String),
+    ScriptHashChange(FullHash, Option<FullHash>, bool),
```

Comment:
I think that if the 2nd parameter (the txid) of type `Option<FullHash` is `None` - then it is a backfill.
So I guess you can drop the 3rd parameter (the boolean for is_backfill).